src/block/Lantern.php

```diff
@@ -72,22 +73,27 @@ protected function recalculateCollisionBoxes() : array{
 		];
 	}
 
-	protected function canAttachTo(Block $b) : bool{
-		return !$b->isTransparent();
-	}
-
 	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
-		if(!$this->canAttachTo($this->position->getWorld()->getBlock($blockReplace->getPosition()->up())) && !$this->canAttachTo($this->position->getWorld()->getBlock($blockReplace->getPosition()->down()))){
+		if(Facing::axis($face) === Axis::Y && !$this->canBeSupportedBy($blockReplace->getSide(Facing::opposite($face)), $face)){
```

Comment:
This is incorrect. The block should look for up/down supports regardless of which face was clicked.

You can see the problem by right-clicking with a lantern on a wall when there are no available attachment points - the lantern item will be consumed, but no lantern is placed.